### PR TITLE
Use OpenAPI spec to document API

### PR DESCRIPTION
**Pull request by gravesm**
#### What does this PR do?

This adds an ADR for using OpenAPI. Closes
https://mitlibraries.atlassian.net/browse/DIP-98.

#### What are the relevant tickets?

- https://mitlibraries.atlassian.net/browse/DIP-98

#### Requires Full Reindexing of all Sources?
NO

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: docs/architecture-decisions/0010-use-openapi-specification.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+By choosing an API documentation standard we make it easier to auto generate developer documentation. There are two existing standards used to document REST APIs--RAML and OpenAPI Specification (OAS). Both seem capable of doing the job. The main difference seems to be that RAML is focused on defining data models while OpenAPI is focused on the nuts and bolts of the API. If we were supporting several APIs, RAML might be more useful for defining reusable types across systems. In this case OAS seems more suited to our task.
+
+## Decision
+
+Use OpenAPI specification to document the API.
+
+## Consequences
+
+Given that Mulesoft uses RAML and Swagger uses OAS, this means we'd be probably be using Swagger as a documentation platform (if we choose to do so). There are tools to convert between RAML and OAS, though, so if we decide we would rather use RAML later it should not be too difficult to switch.
```
Comment (JPrevost) [inline docs/architecture-decisions/0010-use-openapi-specification.md]: at first I was worried about this but it gets waffly enough to let it slide ;)

**Comment by JPrevost** [comment]
I noticed this doesn't declare the version of OpenAPI we'll be using which I think is good, but wanted to bring that point to discussion to make sure we agree it isn't an oversight.

**Comment by gravesm** [comment]
Well, I guess it is kind of an oversight in that I just assumed we'd use the latest version. Is there a reason we shouldn't?

**Comment by gravesm** [comment]
:+1:

**Pull request merged**